tldw_Server_API/app/core/Agent_Client_Protocol/adapters/mcp_adapter.py

```
@@ -0,0 +1,179 @@
+        await self._emit(AgentEventKind.STATUS_CHANGE, {"from_status": "idle", "to_status": "working"})
+
+        try:
+            orchestration = pc.get("mcp_orchestration", "agent_driven")
+            if orchestration == "llm_driven":
+                llm_caller = pc.get("llm_caller")
+                tool_gate = pc.get("tool_gate")
+                if llm_caller is None or tool_gate is None:
+                    raise ValueError(
+                        "MCP llm_driven orchestration requires protocol_config keys: "
+                        "llm_caller and tool_gate"
+                    )
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)